packages/matplotlib/src/browser_backend.py

```diff
@@ -1,8 +1,67 @@
 import math
+from typing import Any, Callable
 
 from matplotlib.backend_bases import FigureCanvasBase, NavigationToolbar2, TimerBase
 
-from js import document, window
+from js import clearInterval, clearTimeout, document, setInterval, setTimeout
+from pyodide import JsProxy, create_once_callable, create_proxy
+
+try:
+    from js import devicePixelRatio as DEVICE_PIXEL_RATIO
+except ImportError:
+    DEVICE_PIXEL_RATIO = 1
+
+
+EVENT_LISTENERS: dict[tuple[JsProxy, str, JsProxy], JsProxy] = {}
+
+
+def add_event_listener(elt: JsProxy, event: str, listener: Callable[[Any]]):
```

Comment:
Can you make all of these private, so we don't have to care about backward compatibility? `_add_event_listener` etc.